src/Kestrel.Core/Internal/Http/HttpProtocol.cs

```diff
@@ -607,7 +607,7 @@ private async Task ProcessRequests<TContext>(IHttpApplication<TContext> applicat
                 application.DisposeContext(httpContext, _applicationException);
 
                 // Even for non-keep-alive requests, try to consume the entire body to avoid RSTs.
-                if (_ioCompleted == 0 && _requestRejectedException == null && !messageBody.IsEmpty)
```

Comment:
I think this is trying to avoid a secondary exception if the connection has already closed. Don't merge this until @halter73 can take a look.

Reply:
Yep, I'll need to put this back and probably just add something special for HTTP2 here. It seems like it may have broken something else.

Reply:
There's another way to fix it.